app/src/main/java/me/hufman/androidautoidrive/phoneui/fragments/SpotifyApiErrorDialog.kt

```diff
@@ -1,18 +1,76 @@
 package me.hufman.androidautoidrive.phoneui.fragments
 
+import android.app.Activity
 import android.app.AlertDialog
 import android.app.Dialog
+import android.content.Intent
 import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Button
+import android.widget.TextView
 import androidx.fragment.app.DialogFragment
 import me.hufman.androidautoidrive.R
-import java.lang.IllegalStateException
+import me.hufman.androidautoidrive.phoneui.SpotifyAuthorizationActivity
+import me.hufman.androidautoidrive.phoneui.visible
 
 class SpotifyApiErrorDialog: DialogFragment() {
 	companion object {
+		const val TAG = "SpotifyApiErrorDialog"
 		const val EXTRA_CLASSNAME = "classname"
 		const val EXTRA_MESSAGE = "message"
+		const val EXTRA_WEB_API_AUTHORIZED = "webApiAuthorized"
 	}
+
+	lateinit var webApiMsgTextView: TextView
+	lateinit var authorizeButton: Button
+
 	override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+		val builder = AlertDialog.Builder(activity)
+		val view = activity?.layoutInflater?.inflate(R.layout.dialog_error, null)!!
+		builder.setView(view)
+
+		webApiMsgTextView = view.findViewById(R.id.txtWebApiAuthorizationMsg)
+		authorizeButton = view.findViewById(R.id.btnAuthorizeWebApi)
+
+		updateAppRemoteErrorComponents(view)
+		updateWebApiAuthorizationComponents()
+
+		view.findViewById<Button>(R.id.btnOk)?.setOnClickListener { dismiss() }
+
+		return builder.create()
+	}
+
+	override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
```

Comment:
The demo video shows this SpotifyApiErrorDialog popping up for the auth success, but I'm not actually sure how it's triggered. I'd expect the `onActivityResult` of the SpotifyAuthorizationActivity to be called with the result, not this one directly

Reply:
Oh I see, `SpotifyApiErrorDialog` can trigger `SpotifyAuthActivity` and receive a result and updates its own label appropriately...
I think if the result is Success, it should close the Dialog after a few seconds automatically (`postDelayed {finish()}`) to reduce the steps after the user initiated the successful flow.

Reply:
Yeah I needed some sort of way to process the status of the SpotifyAuthActivity so the result of that authorization activity is packaged with an extra with the appropriate AUTHORIZATION_SUCCESS, AUTHORIZATION_FAILED, etc. Ah that's a good idea, added some logic to close the dialog after 2 seconds automatically